### DISCUSSION TITLE
ColorAnimation throws an exception at load time

### DISCUSSION THREAD

**Discussion by gmgallo** [CommunityToolkit/Windows#292] [Q&A] [answered]
I'm trying to use these implicit animations in a ToggleButton but animations:ColorAnimation throws an exception at load time.
This is the xaml code:
 ```
<ToggleButton x:Name="tbStarStopLog" Height="50" Width="50" IsEnabled="True" 
         Checked="tbStarStopLog_Checked" Unchecked="tbStarStopLog_Unchecked">
             <Image x:Name="imgStartStopLog" Source="/Assets/Circle_Hourglass-48.png"/>
         <animations:Implicit.Animations>
             <animations:OffsetAnimation Duration="0:0:1"/>
             <animations:RotationInDegreesAnimation ImplicitTarget="Offset" Duration="0:0:1.5">
                 <animations:ScalarKeyFrame Key="1.0" Expression="this.Target.Offset.X"/>
             </animations:RotationInDegreesAnimation>
             <animations:ScaleAnimation Duration="0:0:1"/>
                      
                        <animations:RotationInDegreesAnimation Duration="0:0:1"/>

             <!--<animations:ColorAnimation Duration="0:0:1"/>-->
         </animations:Implicit.Animations>
     </ToggleButton>
```
I copied these animations from the toolkit's implicit animations example and added the ColorAnimation (commented out here)
The exception is probably because a Target or other property is missing but I can't find any documentation about it.

I also want to use a simple rotation animation, the one above the ColorAnimation, but that doesn't trigger, the contorl rotates at full speed around the top left corner.

The other thing is that I can't find the correct way to change the rotation origin to the center of the control.
Here is the code behind that triggers the rotation:
   
    private void ButtonRotate_Click(object sender, Microsoft.UI.Xaml.RoutedEventArgs e)
    {
        var visual2 = ElementCompositionPreview.GetElementVisual(tbStarStopLog);
        tbStarStopLog.RenderTransformOrigin = new Windows.Foundation.Point (0.5, 0.5); // this is not the correct way to change it
        visual2.RotationAngle += angle;
    }

**Comment by niels9001** [comment]
@gmgallo `ColorAnimation` is something that isn't part of the Toolkit, right? Is that a custom variant you created?

Wrt the centerpoint: you can use the `VisualExtensions` from the Toolkit to set the CenterPoint in XAML, like this:

`<ToggleButton ui:VisualExtensions.NormalizedCenterPoint="0.5, 0.5" />` See https://github.com/CommunityToolkit/Windows/blob/bef863ca70bb1edf8c940198dd5cc74afa5d2aab/components/Extensions/samples/VisualExtensions.md

**Comment by gmgallo** [comment]
Hi Niels, I didn't create that ColorAnimation.  I thought that it was part of  the toolkit because it pops up as one of the options in Visual Studio when you start typing the class name in the xaml editor. It may be part of the Framework then, but the help on that comes empty.

On the RotationInDegreesAnimation, why it doesn't trigger? should it be declared in some other way ?

Tanks for the hint on how to relocate the origin. It takes time to learn all the ins and outs of this great toolkit.

One implicit animations I would like to suggest one that triggers when the control **content** changes. 

I'm replacing Icons on that togglebutton depending on the current state of the application (hourglass, start, stop,etc.), but havent found an elegant way to do it in Xaml alone, and don't have the time to develop a custom control just for that, so, I do it quick and dirty on the code behind. Any suggestions for this?

**Reply by niels9001** [reply, marked as answer]
`ColorAnimation`: yeah, I believe it's part of the platform. Check the `Animations` section in the WinUI Gallery, there are some things listed there `BrushTransition` could be of help?

`RototationInDegreesAnimation`: https://github.com/CommunityToolkit/Windows/blob/bef863ca70bb1edf8c940198dd5cc74afa5d2aab/components/Animations/samples/AnimationsImplicitSample.xaml might be of help

Showing content: `SwitchPresenter` could be useful for this: https://github.com/CommunityToolkit/Windows/blob/bef863ca70bb1edf8c940198dd5cc74afa5d2aab/components/Primitives/samples/SwitchPresenter/SwitchPresenterValueSample.xaml

**Reply by gmgallo** [reply]
Thanks Niels!
